CommandLine/testing/Platform/TestHarness-X11.cpp

```diff
@@ -0,0 +1,212 @@
+#include "TestHarness.hpp"
+
+#include <cstdint>
+#include <cstdio>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <sys/wait.h>
+#include <unistd.h>
+#include <X11/Xlib.h>
+#include <X11/Xutil.h>
+
+namespace {
+using std::string;
+using std::to_string;
+using std::unique_ptr;
+
+string get_window_caption(Display *disp, Window win) {
```

Comment:
TABS! REEEEEEE

Reply:
Removed. This is the problem with copying code from elsewhere in the engine. 😛